File: sherpout-server/src/main/java/com/sherpout/server/error/model/ErrorLocationType.java
package com.sherpout.server.error.model;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum ErrorLocationType {

    PATH_PARAM("Path Parameter"),
    QUERY_PARAM("Query Parameter"),
    HEADER_PARAM("Header Parameter"),
    BODY("Body"),
    FILE("File");

    private final String name;
}

File: sherpout_mobile/lib/common/api/error/error_location_type.dart
enum ErrorLocationType {
  pathParam,
  queryParam,
  headerParam,
  body,
  file;

  static ErrorLocationType? fromName(String name) {
    return ErrorLocationType.values.firstWhere(
      (e) => e.name.toUpperCase() == name.replaceAll("_", ""),
    );
  }
}
